server/src/main/java/org/tctalent/server/service/db/impl/SavedSearchServiceImpl.java

```diff
@@ -361,29 +361,32 @@ public Page<Candidate> searchCandidates(SearchCandidateRequest request) {
         Page<Candidate> candidates;
         User user = userService.getLoggedInUser();
         if (user == null) {
+            // todo when is the user null?
             candidates = doSearchCandidates(request);
         } else {
-            //Update default search
-            SavedSearch defaultSavedSearch =
-                getDefaultSavedSearch();
-            Long savedSearchId = defaultSavedSearch.getId();
-            UpdateSavedSearchRequest updateRequest = new UpdateSavedSearchRequest();
-            updateRequest.setSearchCandidateRequest(request);
-            //Set other fields - no changes there
-            updateRequest.setName(defaultSavedSearch.getName());
-            updateRequest.setDefaultSearch(defaultSavedSearch.getDefaultSearch());
-            updateRequest.setFixed(defaultSavedSearch.getFixed());
-            updateRequest.setReviewable(defaultSavedSearch.getReviewable());
-            updateRequest.setSavedSearchType(defaultSavedSearch.getSavedSearchType());
-            updateRequest.setSavedSearchSubtype(defaultSavedSearch.getSavedSearchSubtype());
-            //todo Need special method which only updates search part. Then don't need the above "no changes there" stuff
-            updateSavedSearch(savedSearchId, updateRequest);
+            SavedSearch savedSearch = getSavedSearch(request.getSavedSearchId());
+            // If searching a default search, update the default search with every search.
+            // Else it is a saved search and those are updated upon 'Update Search' button only.
+            if (savedSearch.getDefaultSearch()) {
+                UpdateSavedSearchRequest updateRequest = new UpdateSavedSearchRequest();
+                updateRequest.setSearchCandidateRequest(request);
+                //Set other fields - no changes there
+                updateRequest.setName(savedSearch.getName());
+                updateRequest.setDefaultSearch(savedSearch.getDefaultSearch());
+                updateRequest.setFixed(savedSearch.getFixed());
+                updateRequest.setReviewable(savedSearch.getReviewable());
+                updateRequest.setSavedSearchType(savedSearch.getSavedSearchType());
+                updateRequest.setSavedSearchSubtype(savedSearch.getSavedSearchSubtype());
+                //todo Need special method which only updates search part. Then don't need the above "no changes there" stuff
+                updateSavedSearch(savedSearch.getId(), updateRequest);
+            }
 
             //Do the search
             candidates = doSearchCandidates(request);
 
             //Add in any selections
-            markUserSelectedCandidates(savedSearchId, candidates);
+            // todo do we want to remove a selection when it get's removed from a search?
```

Comment:
Related to discussion around selections for unsaved searches - if filters are changed and a selected candidate no longer shows up in search results, could it be unselected?